### PR TITLE
Added supporting hitl testing for the mavsdk_test_runner

### PR DESCRIPTION
**Pull request by Ilia-Loginov**
Support hitl testing for the mavsdk_test_runner includes:
 - Special airframe
 - Adding an application to set up a test device
 - Updating a test configuration
 - Updating the runner to support hitl mode and gz sim

### CODE REVIEW

File: test/mavsdk_tests/mavsdk_test_runner.py [hunk 1]
```diff
@@ -209,6 +210,150 @@ def run(self) -> bool:
...
+        if self.connection == "serial":
+            res= self.check_connection_px4_serial()
+
+        elif self.connection == "ethernet":
+            res = self.check_connection_px4_ethernet()
+
+        if res:
+            print("Device has ", self.connection, " connection")
+        else:
+            print("Device doesn't has ",self.connection, " connection")
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_test_runner.py:343]:
```suggestion
            print("Device doesn't have ",self.connection, " connection")
```

File: test/mavsdk_tests/mavsdk_test_runner.py [hunk 2]
```diff
@@ -209,6 +210,150 @@ def run(self) -> bool:
...
+
+    def reboot_px4(self):
+        time.sleep(5)
+        res = False
+        if (self.connection == "serial"):
+            res = self.reboot_using_serial()
+        elif (self.connection == "ethernet"):
+            res = self.reboot_using_ethernen()
+        else:
+            print("Chose incorrect connection")
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_test_runner.py:316]:
```suggestion
            print("Wrong connection selected")
```

File: test/mavsdk_tests/mavsdk_test_runner.py [hunk 3]
```diff
@@ -209,6 +210,150 @@ def run(self) -> bool:
...
+
+        if (res):
+            print("The device has successfully rebooted")
+            time.sleep(15)
+        else:
+            print("Rebooting failed")
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_test_runner.py:322]:
```suggestion
            print("Reboot failed")
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		return 1;
+	}
+
+	std::cout << "Waiting for system connect" << std::endl;
+
+
+	auto res_polling = poll_condition_with_timeout([&]() { return mavsdk.systems().size() > 0;}, std::chrono::seconds(25));
+
+	if (not res_polling) {
+		std::cerr << "Polling was failed" << std::endl;
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:104]:
```suggestion
		std::cerr << "Polling failed" << std::endl;
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+		return 1;
+	}
+
+	auto system = mavsdk.systems().at(0);
+
+	if (!system) {
+		std::cerr << "The system wasn't be found" << std::endl;
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:111]:
```suggestion
		std::cerr << "The system could not be found" << std::endl;
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+				} else {
+					command.erase();
+				}
+			}
+		}
+
+	}
+
+	if (connection_url.empty()) {
+		std::cerr << "No connection URL  was supplied" << std::endl;
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:86]:
```suggestion
		std::cerr << "Connection URL not provided" << std::endl;
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+		const int max_attemp = 5;
+		const auto param_name = "SYS_AUTOSTART";
+
+		for (attempt = 0; attempt <= max_attemp; attempt++) {
+			using namespace std::chrono_literals;
+			auto param = Param(system);
+
+
+			if (param.set_param_int(param_name, param_value) != Param::Result::Success) {
+				std::cerr << "Fail setting param: " << param_name << ", value: "
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:132]:
```suggestion
				std::cerr << "Fail setting parameter: " << param_name << ", value: "
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+			if (res.first == Param::Result::Success && res.second == param_value) {
+				break;
+			}
+
+			std::this_thread::sleep_for(1s);
+		}
+
+		if (attempt <= max_attemp) {
+			std::cout << "Param " << param_name << " was successly set after " << attempt
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:147]:
```suggestion
			std::cout << "Parameter " << param_name << " was successly set after " << attempt
```

File: test/mavsdk_tests/mavsdk_preparing.cpp [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+				  << " attempt. Tne new value " << param_value << std::endl;
+
+		} else {
+			std::cerr << "Fail. Setting param: " << param_name << " wasn't set to value: "
```
Comment (Ilia-Loginov) [inline test/mavsdk_tests/mavsdk_preparing.cpp:151]:
```suggestion
			std::cerr << "Fail. Setting parameter: " << param_name << " wasn't set to value: "
```

**Review by haitomatic — APPROVED** [review]
i suppose the 1404 vtol airframe  is coming latter after we get it work. otherwise lgtm!

**Review by aravindraja93 — APPROVED** [review]
Looks Good.

**Pull request merged**